From c7643d2a9505cdd5a968815d187c7746891b566c Mon Sep 17 00:00:00 2001
From: O01eg <o01eg@yandex.ru>
Date: Tue, 14 Mar 2023 21:09:26 +0300
Subject: [PATCH] Workaround for boost 1.81 issue

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 67ad3895396..2fa295ed5a0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -346,6 +346,8 @@ set_property(DIRECTORY APPEND
         $<$<PLATFORM_ID:FreeBSD>:BOOST_DATE_TIME_NO_LOCALE>
         # Forcing backtrace using api for boost stacktrace due to lacking on ::Unwind side
         $<$<PLATFORM_ID:FreeBSD>:BOOST_STACKTRACE_USE_LIBC_BACKTRACE_FUNCTION>
+        # Workaround for boost 1.81 issue https://github.com/boostorg/phoenix/issues/111
+        $<$<VERSION_EQUAL:${Boost_MAJOR_VERSION}.${Boost_MINOR_VERSION},1.81>:BOOST_PHOENIX_STL_TUPLE_H_>
 )
 
 
