diff -uraN freeorion-0.5.1.1/CMakeLists.txt omv-freeorion-0.5.1.1/CMakeLists.txt
--- freeorion-0.5.1.1/CMakeLists.txt	2025-06-15 19:43:26.000000000 +0200
+++ omv-freeorion-0.5.1.1/CMakeLists.txt	2025-11-08 16:33:17.243777314 +0100
@@ -283,7 +283,6 @@
         log
         log_setup
         serialization
-        system
         thread
     REQUIRED)
 
diff -uraN freeorion-0.5.1.1/GG/CMakeLists.txt omv-freeorion-0.5.1.1/GG/CMakeLists.txt
--- freeorion-0.5.1.1/GG/CMakeLists.txt	2025-06-15 19:43:26.000000000 +0200
+++ omv-freeorion-0.5.1.1/GG/CMakeLists.txt	2025-11-08 16:33:49.925680207 +0100
@@ -66,7 +66,7 @@
 
 set(MINIMUM_BOOST_VERSION 1.69.0)
 
-find_package(Boost ${MINIMUM_BOOST_VERSION} COMPONENTS filesystem regex system REQUIRED)
+find_package(Boost ${MINIMUM_BOOST_VERSION} COMPONENTS filesystem regex REQUIRED)
 
 find_package(OpenGL REQUIRED)
 find_package(GLEW REQUIRED)
diff -uraN freeorion-0.5.1.1/GG/test/unit/CMakeLists.txt omv-freeorion-0.5.1.1/GG/test/unit/CMakeLists.txt
--- freeorion-0.5.1.1/GG/test/unit/CMakeLists.txt	2025-06-15 19:43:26.000000000 +0200
+++ omv-freeorion-0.5.1.1/GG/test/unit/CMakeLists.txt	2025-11-08 16:34:12.564259729 +0100
@@ -16,7 +16,6 @@
 target_link_libraries(gg_unittest
     GiGi
     Boost::boost
-    Boost::system
     Boost::unit_test_framework
 )
 
diff -uraN freeorion-0.5.1.1/test/UI/CMakeLists.txt omv-freeorion-0.5.1.1/test/UI/CMakeLists.txt
--- freeorion-0.5.1.1/test/UI/CMakeLists.txt	2025-06-15 19:43:26.000000000 +0200
+++ omv-freeorion-0.5.1.1/test/UI/CMakeLists.txt	2025-11-08 16:35:57.409990748 +0100
@@ -23,7 +23,6 @@
 
 target_link_libraries(fo_acceptance_runner
     PUBLIC
-        Boost::system
         GiGi::GiGi
         GLEW::GLEW
         ${SDL_LIBRARIES}
